src/import/trace-event.ts

```diff
@@ -100,15 +112,21 @@ function selectQueueToTakeFromNext(
   if (bts < ets) return 'B'
   if (ets < bts) return 'E'
 
-  // If we got here, the 'B' event queue and the 'E' event queue have events at
-  // the front with equal timestamps.
+  // If the keys are the same then we need to check whether the number of begin events
+  // with that key are the same as the number of end events with that key for the given timestamp.
+  // - If the number of begin and end events are the same, then we have a zero duration event,
+  // and we should process the begin first
+  // - In all other cases we should process the end event first
```

Comment:
I'm having trouble following this logic. If you had, conceptually:

```
B funcX ts=1
B funcY ts=1
E funcX ts=1
E funcY ts=2
```

Then the correct processing order is:

- B funcX ts=1
- E funcX ts=1 (a zero-duration event)
- B funcY ts=1
- E funcY ts=2

But if I'm reading the description here correctly, this code would process in the following order, because the number of begin events and end events at the given timestamp are different:

- E funcX ts=1
- B funcX ts=1
- B funcY ts=1
- E funcY ts=2

Am I misunderstanding?

Reply:
Sorry I didn't realized I pushed this up - this is not quite right. For now, I'm going to push up the original change, which just uses keys to avoid this logic for my case and is better. I'll make a separate PR to fix the general case, which is a bit more complicated and I need to think about some more